imxrt-hal/src/srtc.rs

```diff
@@ -0,0 +1,144 @@
+//! (Secure) Real-Time Clock
+//!
+//! Basic support only: Supports enabling and setting the clocks, but not any of their more
+//! advanced features (alarm, interrupt, power glitch detection, etc).
+//!
+//! The Secure Real-Time Clock continues tracking time until it is specifically disabled or loses
+//! power, even through a system reboot (and potentially a loss of power to the system, if the
+//! SRTC is connected to a battery).
+//!
+//! # Example
+//!
+//! ```no_run
+//! use imxrt_hal;
+//!
+//! let mut peripherals = imxrt_hal::Peripherals::take().unwrap();
+//!
+//! let mut rtc = peripherals.rtc.enable_and_set(1600000000);
+//! // Interpreted as Unix time: Sep 13 2020 12:26:40
+//!
+//! let now = rtc.get();
+//! ```
+//!
+//! # Teensy 4.x Note
+//!
+//! When the SRTC is enabled, setting the board into program mode then using the Teensy Loader
+//! to reboot it will set the current time (Unix epoch, but time in local timezone). This will
+//! overwrite whatever time you may have previously set.
+
+use core::fmt;
+
+use crate::ral;
+use crate::ral::snvs::Instance;
+
+const MR_SHIFT: u8 = 17;
+const LR_SHIFT: u8 = 15;
+
+/// A disabled RTC.
+///
+/// Note: The SRTC may actually be enabled, as (unless disabled) it stays enabled across reboots
+/// as long as it has power.
+pub struct Unclocked {
+    reg: Instance,
+}
```

Comment:
Just a note that the CCM clock gating register 5 (CCM_CCGR5) has two clock gate fields for the SNVS high- and low-power domains. The reset values are 'run'. According to "Clocks at Boot Time" section (table 9-7, iMXRT1060 reference manual, rev 2), the boot ROM does not disable these clocks.

We'll note that some other HAL drivers explicitly enable their clock gates, even if they are not disabled by boot ROM. We might consider doing the same here, but it would just be for consistency. We're OK without it!

Reply:
Added clock gate enable using the CCM handle.